src/mlpack/core/data/split_data.hpp

```diff
@@ -224,6 +229,102 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+/**
+ * Performs train test split.
+ *
+ * @tparam DatasetType Type of input dataset.
+ * @tparam LabelsType Type of input labels.
+ *
+ * @param dataset Features of dataset.
+ * @param labels Labels of the dataset.
+ * @param validRatio Ratio for train-test split.
+ * @param shuffle Boolean to determine shuffling of dataset.
+ */
+template<typename DatasetX>
+void Split(DatasetX& dataset,
+                    std::deque<arma::vec>& labels,
+                    DatasetX& trainFeatures,
+                    arma::field<arma::vec>& trainLabels,
+                    DatasetX& validFeatures ,
+                    arma::field<arma::vec>& validLabels,
+                    const double validRatio,
+                    const bool shuffle)
```

Comment:
In mlpack/mlpack, it will be testRatio.

Reply:
Thanks, I wanted to focus on the working first. Have you any suggestion on the implementation?